classic-editor.php

```diff
@@ -935,6 +945,21 @@ public static function uninstall() {
 		delete_option( 'classic-editor-replace' );
 		delete_option( 'classic-editor-allow-users' );
 	}
+
+	/**
+	 * Get a summary of settings for a given user id.
+	 */
+	public static function get_user_settings( $user_id ) {

```

Comment:
We can potentially use the existing private internal get_settings method here instead of get_options calls for some of these

Reply:
Maybe we could also include the post_id in the call and cover off some of those edge cases for specific post editor settings @Copons ?

Reply:
This summary already covers a lot.
Details we are missing (regardless of our intention to use them):
- The last editor used on a post.
- The editors enabled for a post.
- The editors enabled for a post type.
- The network options on multisites.

The last editor used on a post is saved in the `classic-editor-remember` post meta.
I don't think we should add this information here, to be honest, but we could explicitly register it to make it show up in the `/wp/v2/posts` (`edit` and `view` contexts) response.
E.g.
```php
register_meta( 'post', 'classic-editor-remember', [
  'show_in_rest' => true,
  'single' => true,
] );
```

If a post type is registered without `editor` support, it will always use the "classic editor UI".
This info returns in `/wp/v2/types` (but only in `edit` context, meaning it needs to be authorized to edit post types).

Then, there are two filters affecting the list of available editor for any given post type and even post:
```php
apply_filters( 'classic_editor_enabled_editors_for_post_type', $editors, $post_type );
apply_filters( 'classic_editor_enabled_editors_for_post', $editors, $post );
```

Since post types are generaly a finite number, I guess we could return a map of available editors for each of them. Something like:
```js
{
  post: [ 'classic', 'block' ],
  page: [ 'block' ],
}
```
But, given these options are not tied to users, I'd rather not conflate them into the user endpoint, and create a new one(s) to query editors availability per (single) post and post type.

Last but not least: the network settings.
They are already considered in the plugin logic.
I'm about to push a commit that reuses the plugin's original `get_settings` method to take advantage of all its niceties!